### PR TITLE
Fix/omemo

### PR DESCRIPTION
**Pull request by melvo**
PR check list:
- [x] Document your code
- [x] Add `\since QXmpp 1.X`, `QXMPP_EXPORT`
- [x] Fix doxygen warnings (see log when building with `-DBUILD_DOCUMENTATION=ON`)
- [x] Update `doc/doap.xml`
- [x] Add unit tests
- [x] Format the code: Run `clang-format -i src/<edited-file(s)> tests/<edited-file(s)>`

### CODE REVIEW

**Comment by lnjX** [comment]
breaks qt6 build

File: src/omemo/QXmppOmemoManager_p.cpp [hunk 1]
```diff
@@ -2650,6 +2650,36 @@ void ManagerPrivate::updateOwnDevicesLocally(bool isDeviceListNodeExistent, Func
...
+// \param deviceOwnerJid bare JID of the devices' owner
+// \param deviceListItems PEP items that may contain a device list
+//
+// \returns a found device list item
+//
+std::optional<QXmppOmemoDeviceListItem> QXmppOmemoManagerPrivate::updateContactDevices(const QString &deviceOwnerJid, const QVector<QXmppOmemoDeviceListItem> &deviceListItems)
+{
+    if (deviceListItems.size() > 1) {
+        const auto itr = std::find_if(deviceListItems.cbegin(), deviceListItems.cend(), [=](const QXmppOmemoDeviceListItem &item) {
+            return item.id() == QXmppPubSubManager::Current;
```
Comment (lnjX) [inline src/omemo/QXmppOmemoManager_p.cpp:2665]: how is this supposed to work?

**Comment by melvo** [comment]
> breaks qt6 build

It seems that older commits (e.g., https://github.com/qxmpp-project/qxmpp/commit/366a22985b946b70252ba5e6dc291dbfe8addd94) already result in failing Qt 6 builds. What exactly do you mean?

File: src/omemo/QXmppOmemoManager_p.cpp (hunk 1 above)
Comment (lnjX) [reply src/omemo/QXmppOmemoManager_p.cpp:2665]: this broke qt6

**Pull request merged**